include/oneapi/dpl/pstl/numeric_impl.h

```diff
@@ -217,7 +212,8 @@ __brick_transform_scan(_RandomAccessIterator __first, _RandomAccessIterator __la
 
 template <class _RandomAccessIterator, class _OutputIterator, class _UnaryOperation, class _Tp, class _BinaryOperation,
           class _Inclusive>
-typename ::std::enable_if<is_arithmetic_udop<_Tp, _BinaryOperation>::value, ::std::pair<_OutputIterator, _Tp>>::type
+typename ::std::enable_if<!oneapi::dpl::__unseq_backend::is_arithmetic_plus<_Tp, _BinaryOperation>::value,
```

Comment:
And the previous check `"is_arithmetic_udop<_Tp, _BinaryOperation>::value"` must prevent from an error when we have arithmetic type and not "plus" binary operation. Is not it so?

In ideal, we have to dispatch the vectorized  "brick" to several code branches to process the all binary operations supported by OpenMP  spec:

![image](https://user-images.githubusercontent.com/40659946/213529990-911d8693-9dc6-4e12-ba8e-24bdd15cc126.png)

Reply:
Good point.  It should be valid for more than just arithmetic types and plus according to this table, and we could expand the checks to send to __simd_scan in those cases.
You are correct that for arithmetic types and not "plus" binary operation, it prevents an issue.

However, for non-arithmetic type, it always calls __simd_scan (for supporting compilers), whether or not the default constructor is the identity.  This is reasonable if it is listed this as a known limitation, but I don't see that in the current guide or release notes.